bigchaindb/common/memoize.py

```diff
@@ -0,0 +1,42 @@
+import functools
+import codecs
+from functools import lru_cache
+
+
+class HDict(dict):
+    def __hash__(self):
+        return  int.from_bytes(codecs.decode(self['id'], 'hex'), 'big')
+
+
+@lru_cache(maxsize=16384)
+def from_dict(func, *args, **kwargs):
+    return func(*args, **kwargs)
+
+
+def memoize_from_dict(func):
+
+    @functools.wraps(func)
+    def memoized_func(*args, **kwargs):
+        print(args)
+        new_args = (args[0], HDict(args[1]), args[2])
```

Comment:
That's difficult to understand, can you please add some comments around this code (and the equivalent for `memoize_to_dict`.